sky/packages/sky/lib/src/widgets/page.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'navigator2.dart';
+import 'overlay.dart';
+import 'transitions.dart';
+
+abstract class TransitionRoute extends Route {
+  bool get opaque => true;
+
+  PerformanceView get performance => _performance?.view;
+  Performance _performance;
+
+  Duration get transitionDuration;
+
+  Performance createPerformance() {
+    Duration duration = transitionDuration;
+    assert(duration != null && duration >= Duration.ZERO);
+    return new Performance(duration: duration, debugLabel: debugLabel);
+  }
+
+  dynamic _result;
+
+  void _handleStatusChanged(PerformanceStatus status) {
+    if (status == PerformanceStatus.completed && opaque) {
+      bottomEntry.opaque = true;
+    } else if (status == PerformanceStatus.dismissed) {
+      super.remove(_result);
+    }
+  }
+
+  void add(OverlayState overlayer, OverlayEntry insertionPoint) {
```

Comment:
everywhere else, you call the OverlayState an "overlay" rather than "overlayer".